src/core/include/openvino/core/descriptor/tensor.hpp

```diff
@@ -45,13 +45,17 @@ void set_ov_tensor_legacy_name(Tensor& tensor, const std::string& tensor_name);
 /// \brief Compile-time descriptor of a first-class value that is a tensor.
 class OPENVINO_API Tensor {
 public:
-    Tensor(const element::Type& element_type, const PartialShape& pshape, const std::string& name = "");
+    Tensor(const element::Type& element_type,
+           const PartialShape& pshape,
+           const std::unordered_set<std::string>& names = {});
+    OPENVINO_DEPRECATED("This constructor is deprecated. Please use constructor with set of names")
+    Tensor(const element::Type& element_type, const PartialShape& pshape, const std::string& name);
     Tensor(const element::Type& element_type, const PartialShape& pshape, Node* node, size_t node_output_number);
 
     Tensor(const Tensor&) = delete;
     Tensor& operator=(const Tensor&) = delete;
 
-    std::string get_any_name() const;
+    const std::string& get_any_name() const;
```

Comment:
API is changed. The compilation can be broken for an user.

Reply:
Before we always returned the copy, now we return the const reference. It cannot break the compilation of user application, if user used old API, we will create a copy of the returned object on user side.